Libraries/src/Amazon.Lambda.AspNetCoreServer/Internal/Utilities.cs

```diff
@@ -263,5 +264,20 @@ internal static async Task CopyToAsync(Stream source, Stream destination, long?
             // Double-escape any %2F (encoded / characters) so that they survive URL decoding the path.
             .Replace("%2F", "%252F")
             .Replace("%2f", "%252f"));
+
+        internal static X509Certificate2 GetX509Certificate2FromPem(string clientCertPem)
+        {
+            if (!clientCertPem.StartsWith("-----BEGIN CERTIFICATE-----") || clientCertPem.EndsWith("-----END CERTIFICATE-----"))
+            {
+                throw new InvalidOperationException(
```

Comment:
Wasn't sure which Exception to use (please advise...). Didn't add test coverage here, let me know if desired. (TBH, will APIGW ever send an invalid ClientCert PEM?)

Reply:
I'm fine with the exception type. Mostly it was the error message I cared about. What I find is people call this library via some unit test with incomplete sample requests. I have learn the hard way that when I give NPE or index out of bound errors for incomplete sample requests it throws people off.